controllers/installation_controller.go

```diff
@@ -436,3 +445,15 @@ func (r *InstallationReconciler) applyDeletionPolicy(ctx context.Context, log lo
 	inst.SetAnnotations(annotations)
 	return r.Update(ctx, inst)
 }
+
+func createPorterGRPCClient(ctx context.Context) (porterv1alpha1.PorterClient, error) {
+	conn, err := grpc.DialContext(ctx, "porter-grpc-service:3001", grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, fmt.Errorf("error setting up listener for porter grpc client")
+	}
+	defer conn.Close()
```

Comment:
This is going to close once the function is over, we need to move this defer to the method return and close from there.